recipe/patch_yaml/numpy2_known_lower_bound.yaml

```diff
@@ -63,8 +63,7 @@ then:
 ---
 if:
   name: cupy
-  version_lt: 14.0.0
-  has_depends: python >=3.9*
+  version_lt: 13.0.0
```

Comment:
As the `cupy` package recently became a metapackage that is `noarch`, the `python >=3.9` constraint seemed reasonable here. However that only happened with the CuPy 13.0.0 release ( https://github.com/conda-forge/cupy-feedstock/pull/241 )

Previous builds of `cupy` were arch dependent. Also it is the previous `cupy` builds that lacked the `numpy` upper bound

So this drops the `has_depends` on `python` (as if it were a `noarch` package). Further this constrains this patch to pre-13.0.0, which had the single `cupy` package that was arch dependent